### PR TITLE
Add new methods to allow custom headers in each Hudu API Request

### PR DESCRIPTION
**Pull request by craysiii**
Currently one is unable to add custom headers to the request object in a clean manner.

This functionality can be useful in situations where your Hudu instance is hidden behind Cloudflare Zero Trust. In this instance, the developer needs to add Client ID and Client Secret headers to each request to pass authentication through the tunnel.

I've gone ahead and implemented two new methods, `New-HuduCustomHeaders` and `Remove-HuduCustomHeaders` to accommodate this requirement.

Let me know if there are any issues or concerns you have.

### CODE REVIEW

File: HuduAPI/Private/Invoke-HuduRequest.ps1 [hunk 1]
```diff
@@ -61,6 +61,13 @@ function Invoke-HuduRequest {
         'x-api-key' = (New-Object PSCredential 'user', $HuduAPIKey).GetNetworkCredential().Password;
     }
 
+    if (($null -ne $Int_HuduCustomHeaders) -and ($Int_HuduCustomHeaders.Count -gt 0)) {
```
Comment (iqinit) [inline HuduAPI/Private/Invoke-HuduRequest.ps1:64]: I would change the if statement to:
```suggestion
    if (($Script:Int_HuduCustomHeaders | Measure-Object).count -gt 0){
```

That makes sure you read it from the script scope and also makes the count more reliable.

**Review by iqinit — COMMENTED** [review]
N/A

File: HuduAPI/Private/Invoke-HuduRequest.ps1 (hunk 1 above)
Comment (lwhitelock) [inline HuduAPI/Private/Invoke-HuduRequest.ps1:64]:
```suggestion
    if (($Script:Int_HuduCustomHeaders | Measure-Object).count -gt 0){
```
This way it makes sure it reads it from the script scope and will avoid problems if there is only 1 item in Int_HuduCustomHeaders

**Review by lwhitelock — COMMENTED** [review]
Just one comment

**Pull request merged**